### PR TITLE
Gamepad improvements

### LINKED ISSUE #288: Gamepad: Settings to toggle reporting of accel, analog touch
Some PC applications have trouble locating buttons/inputs when a physical input results in multiple software inputs (touchpad -> digital "buttons", analog scale.)

The proposed fix is to add couple of menu items to toggle reporting of the accel and analog touchpad

The settings can be saved to NVS or be temporary

### LINKED ISSUE #289: Gamepad: Draw analog touch bar
Draw the analog input centroid on the screen for the touchpad, in addition to the existing 5 touch-button squares. Can be done either as a thin cursor moving left/right, or as a filled bar similar to accel

### PR DESCRIPTION
**Pull request by Brycey92**
Closes #288.
Closes #289.

### CODE REVIEW

File: main/modes/mode_gamepad.c [hunk 1]
```diff
@@ -71,43 +103,33 @@ void gamepadTouchCb(touch_event_t* evt);
...
-{
-    display_t* disp;
-    hid_gamepad_report_t gpState;
-    hid_gamepad_ns_report_t gpNsState;
-    font_t ibmFont;
-    uint8_t gamepadType;
-    bool isPluggedIn;
-} gamepad_t;
-
+gamepadMenu_t* gm;
```
Comment (AEFeinstein) [inline main/modes/mode_gamepad.c:131]: `gm` and `gamepadToggleSettings` should be members of `gamepad_t`. It saves a tiny bit of RAM and is generally cleaner

Comment (AEFeinstein) [reply main/modes/mode_gamepad.c:131]: And they don't have to be pointers as members of `gamepad_t` either! They'll all get alloc'd and free'd together

File: main/modes/mode_gamepad.c [hunk 2]
```diff
@@ -484,6 +630,34 @@ void gamepadMainLoop(int64_t elapsedUs __attribute__((unused)))
...
+            int32_t center, intensity;
+            if(gamepad->gpState.buttons & ((touchMap[0] | touchMap[1] | touchMap[2] | touchMap[3] | touchMap[4])))
+            {
+                getTouchCentroid(&center, &intensity);
+                // Subtract 2 from TOUCHBAR_WIDTH while scaling so we can draw a 3px-wide cursor centered on center, without covering the box borders
+                center = (TOUCHBAR_WIDTH - 2) * center / 1024 + 1;
+            }
+            else
+            {
+                center = 1;
```
Comment (AEFeinstein) [inline main/modes/mode_gamepad.c:646]: This isn't consistent with the reported value to the PC. When not touched, the reported value is 50%, not 0%. You can verify this with the Windows Game Controller Properties dialog.

**Review by AEFeinstein — CHANGES_REQUESTED** [review]

File: main/modes/mode_gamepad.c (hunk 1 above)
Comment (Brycey92) [reply main/modes/mode_gamepad.c:131]: I can wrap `gm` into `gamepad_t` but I wanted to have `gamepadToggleSettings` separate so I could save and load it as an NVS blob

File: main/modes/mode_gamepad.c (hunk 2 above)
Comment (Brycey92) [reply main/modes/mode_gamepad.c:646]: Ahh this was throwing me off https://github.com/AEFeinstein/Super-2023-Swadge-FW/blob/main/main/modes/mode_gamepad.c#L792

File: main/modes/mode_gamepad.c (hunk 1 above)
Comment (Brycey92) [reply main/modes/mode_gamepad.c:131]: Oh wait I misunderstood you and only realized partway through fixing this. The way I ended up doing it puts all of `gm`'s stuff into `gamepad`, removing `gamepadMenu_t`, and makes `gamepadToggleSettings` a pointer inside `gamepad`

**Review by AEFeinstein — APPROVED** [review]

**Pull request merged**